### PR TITLE
[BC-001] Project bootstrapping

### PR DESCRIPTION
**Pull request by sspr**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,31 +3,52 @@
...
+    "@mui/material": "^5.4.0",
+    "@mui/styled-engine-sc": "^5.3.0",
     "@testing-library/jest-dom": "^5.16.2",
-    "@testing-library/react": "^12.1.2",
     "@testing-library/user-event": "^13.5.0",
     "@types/jest": "^27.4.0",
     "@types/node": "^16.11.22",
     "@types/react": "^17.0.39",
     "@types/react-dom": "^17.0.11",
+    "eslint-plugin-import": "^2.25.4",
```
Comment (szumike) [inline package.json]: About these two - shouldn't they be in `devDependencies`?

File: package.json [hunk 2]
```diff
@@ -3,31 +3,52 @@
...
+    "lint-staged": "^12.3.3",
     "react": "^17.0.2",
     "react-dom": "^17.0.2",
+    "react-intl": "^5.24.4",
     "react-scripts": "5.0.0",
+    "styled-components": "^5.3.3",
```
Comment (szumike) [inline package.json]: Is it necessary to have both, Material UI and styled-components?

File: src/assets/styles/theme.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
+import { createTheme } from '@mui/material/styles';
+import { blue, blueGrey, deepOrange, green, purple, red } from '@mui/material/colors';
+
+declare module '@mui/material/styles' {
```
Comment (szumike) [inline src/assets/styles/theme.ts]: There's nice to have all the types definitions in separate files in `src/types` directory with `*.d.ts` extension

File: src/app/App.tsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import { useIntl } from 'react-intl';
+import Button from '@mui/material/Button';
+
+import { theme } from 'assets/styles/theme';
+
+export const App = () => {
+  const { formatMessage } = useIntl();
```
Comment (szumike) [inline src/app/App.tsx]: `useLocale`, right?

File: src/hooks/useLocale/useLocale.ts [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import { useContext } from 'react';
+import { IntlShape, useIntl } from 'react-intl';
+import { LocaleContext } from 'context/locale/localeContext/LocaleContext';
```
Comment (szumike) [inline src/hooks/useLocale/useLocale.ts:4]: According to the eslint rules, there should be an empty line between external libs imports and the local ones, right? Are you sure, eslint is configured properly?

**Review by szumike — CHANGES_REQUESTED** [review]

File: src/hooks/useLocale/useLocale.ts (hunk 5 above)
Comment (sspr) [reply src/hooks/useLocale/useLocale.ts:4]: I am. When I changed `useIntl` to `useLocale`, it marked me code above as error.

**Review by szumike — APPROVED** [review]

**Pull request merged**